packages/lesswrong/components/comments/CommentPermalink.tsx

```diff
@@ -38,16 +46,23 @@ const CommentPermalink = ({ documentId, post, classes }: {
     fragmentName: 'CommentWithRepliesFragment',
   });
   const refetch = data?.refetch;
-  const { Loading, Divider, CommentWithReplies } = Components;
+  const { Loading, Divider, CommentWithReplies, HeadTags } = Components;
 
   if (error || (!comment && !loading)) return <div>Comment not found</div>
 
   if (!documentId) return null
 
+  const ogUrl = postGetPageUrl(post, true) // open graph
+  const canonicalUrl = post.canonicalSource || ogUrl
+  // For imageless posts this will be an empty string
+  const socialPreviewImageUrl = post.socialPreviewImageUrl
```

Comment:
A bit of redundancy here with `PostsPage`, but after exploring other possibilities this felt like the best choice to me. The other options as I saw it were:

1. pass down a function from `PostsPage` to run this shared logic and return `HeadTags`. This felt like it bound `CommentPermalink` to `PostsPage` in a way that would require some duplication or refactoring if `CommentPermalink` were ever used elsewhere.
2. create a shared component for generating `HeadTags` on posts and comments permalinks. The most elegant option, but it also felt overengineered for only 2 uses and only 4 duplicate lines avoided (and 4 lines which probably won't need modified).
3. access the comment data in `PostsPage`, which requires moving or duplicating the above `useSingle` call to `PostsPage`. Since it's a hook it can't be called conditionally, and it isn't valid if there isn't a `commentId` in the URL. This seemed unworkable.